src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,392 @@ void testNodesHaveIds() throws Exception {
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, filteredRespFuture)) {
+                                filteredRespFuture.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        JsonArray filteredResp = filteredRespFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 2 after filtering recordings by names",
+                filteredResp.size(),
+                Matchers.equalTo(2));
+
+        List<String> filteredRecordingNames = new ArrayList<>();
+        for (int i = 0; i < filteredResp.size(); i++) {
+            JsonObject filteredRecordingInfo = filteredResp.getJsonObject(i);
+            String recordingName = filteredRecordingInfo.getString("name");
+            filteredRecordingNames.add(recordingName);
+            MatcherAssert.assertThat(
+                    filteredRecordingInfo.getString("state"), Matchers.equalTo("RUNNING"));
+        }
+
+        MatcherAssert.assertThat(
+                filteredRecordingNames, Matchers.containsInAnyOrder("Recording1", "Recording2"));
+
+        // Delete recordings
+        for (String recordingName : recordingNames) {
+            CompletableFuture<Void> deleteRecordingFuture = new CompletableFuture<>();
+            webClient
+                    .delete(
+                            String.format(
+                                    "/api/v1/targets/%s/recordings/%s",
+                                    SELF_REFERENCE_TARGET_ID, recordingName))
+                    .send(
+                            ar -> {
+                                if (assertRequestStatus(ar, deleteRecordingFuture)) {
+                                    deleteRecordingFuture.complete(null);
+                                }
+                            });
+            deleteRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        }
+        // Verify no recordings available
+        CompletableFuture<JsonArray> listRespFuture4 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture4)) {
+                                listRespFuture4.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        listResp = listRespFuture4.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 0 after deleting recordings",
+                listResp.size(),
+                Matchers.equalTo(0));
+    }
+
+    @Test
+    @Order(11)
+    public void shouldReturnArchivedRecordingsFilteredByNames() throws Exception {
```

Comment:
Same here, this test exercises a recording creation/archive/deletion workflow, but it doesn't actually use GraphQL to do so.

Reply:
I  saw on the  HTTP_API.md  GET/api/vi/recordings will return all archived recordings then I can just match the names?

Reply:
i got it though... Also saw an example

Reply:
You can, yes, but the point of this class is to exercise the GraphQL endpoint, and this test in particular should exercise the new code you have added. If you use `GET /api/v1/recordings` you will get data back, but you will not be using the GraphQL `names` query that you added, so what is being tested exactly?

Reply:
ok. got it